src/Bicep.Core/Semantics/SemanticModel.cs

```diff
@@ -161,16 +165,27 @@ public IReadOnlyList<IDiagnostic> GetAnalyzerDiagnostics(ConfigHelper? overrideC
         }
 
         /// <summary>
-        /// Gets all the diagnostics sorted by span position ascending. This includes lexer, parser, and semantic diagnostics.
+        /// Cached diagnostics from compilation
         /// </summary>
-        public IEnumerable<IDiagnostic> GetAllDiagnostics(ConfigHelper? overrideConfig = default) =>
-            GetParseDiagnostics()
+        public IEnumerable<IDiagnostic> GetAllDiagnostics(ConfigHelper? overrideConfig = default)
```

Comment:
>overrideConfig [](http://example.com/codeflow?start=72&length=14)

Shouldn't the override be part of the SemanticModel, so there's only one way to get the diagnostics and that's the set that gets cached all the time?

(My understanding is that we only use it for tests, so the overide is always known up front.)

Reply:
If this is complicated, I'm fine not fixing right now and deferring to a later PR.

Reply:
Passing in at compilation time achieve what you are looking for here?

Reply:
Seems pretty hairy to change the configuration in all the compilation infrastructure.

Reply:
It may very well be, but the config file is one of the inputs that affects the outcome of a compilation, so it makes sense it's passed in at the beginning. It's also confusing that the function sometimes caches and sometimes doesn't. On top of that, I think we will need it for these features as well:
- Suppressing core compiler warnings via bicepconfig.json
- Monitoring bicepconfig.json for changes in the language server

But, we don't need it in this PR.